packages/langium/src/utils/promise-util.ts

```diff
@@ -86,34 +86,119 @@ export async function interruptAndCheck(token: CancellationToken): Promise<void>
 }
 
 /**
- * Utility class to execute mutually exclusive actions.
+ * Utility service to execute mutually exclusive actions.
  */
-export class MutexLock {
-
-    private previousAction = Promise.resolve();
-    private previousTokenSource = new CancellationTokenSource();
-
+export interface MutexLock {
```

Comment:
Since this is a workspace-related service with a clear intent, rather than a reusable utility, I'd propose to rename it to `WorkspaceLock` and move the interface and default implementation to `src/workspace/workspace-lock.ts`.